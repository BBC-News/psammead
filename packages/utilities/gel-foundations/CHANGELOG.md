# Gel Foundations Changelog

| Version | Description |
|---------|-------------|
| 0.1.4   | [PR#224](https://github.com/BBC-News/psammead/pull/224) Add tests for the exported values, coverage 100% :tada: |
| 0.1.1   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.0   | [PR#221](https://github.com/BBC-News/psammead/pull/221) Create initial package, pulled in from gel-constants and gel-foundations-styled-components packages. |

