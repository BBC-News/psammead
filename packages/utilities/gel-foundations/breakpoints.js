module.exports = require('./dist/breakpoints'); // eslint-disable-line import/no-unresolved
