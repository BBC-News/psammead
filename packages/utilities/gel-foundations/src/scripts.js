export latin from './scripts/latin-and-cyrillic';
export cyrillic from './scripts/latin-and-cyrillic';
export latinDiacritics from './scripts/latin-with-diacritics';
export arabic from './scripts/arabic';
export arabicPashto from './scripts/arabic-pashto';
export bengali from './scripts/bengali';
export burmese from './scripts/burmese';
export hindi from './scripts/devanagari-and-gurmukhi';
export nepali from './scripts/devanagari-and-gurmukhi';
export sinhalese from './scripts/sinhalese';
export tamil from './scripts/tamil';
export thai from './scripts/thai';
export chinese from './scripts/no-ascenders-or-descenders';
export korean from './scripts/no-ascenders-or-descenders';
export ethiopic from './scripts/no-ascenders-or-descenders';
