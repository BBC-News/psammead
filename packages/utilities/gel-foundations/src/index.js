import * as breakpoints from './breakpoints';
import * as propTypes from './prop-types';
import * as scripts from './scripts';
import * as spacings from './spacings';
import * as typography from './typography';

export { breakpoints };
export { propTypes };
export { scripts };
export { spacings };
export { typography };
