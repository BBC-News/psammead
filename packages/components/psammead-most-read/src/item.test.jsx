import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import Timestamp from '@bbc/psammead-timestamp';
import { latin, arabic } from '@bbc/gel-foundations/scripts';
import { MostReadLink, MostReadRank } from './item';

const ltrLink = {
  title: 'John Lewis staff bonus cut again as profits fall',
  href: 'https://www.bbc.com/vietnamese/institutional-49283563',
};
const rtlLink = {
  title: "ایران از لغو 'رزمایش قطع اینترنت' خبر داد",
  href: 'https://www.bbc.com/vietnamese/institutional-49283563',
};

const lastUpdated = (script, service) => (
  <Timestamp
    datetime="2019-03-01T14:00+00:00"
    script={script}
    padding={false}
    service={service}
  >
    Last updated: 5th November 2016
  </Timestamp>
);

describe('MostReadLink', () => {
  shouldMatchSnapshot(
    'should render ltr correctly',
    <MostReadLink link={ltrLink} service="news" script={latin} dir="ltr" />,
  );

  shouldMatchSnapshot(
    'should render rtl correctly',
    <MostReadLink link={rtlLink} service="persian" script={arabic} dir="rtl" />,
  );

  shouldMatchSnapshot(
    'should render with last updated date correctly',
    <MostReadLink
      link={ltrLink}
      lastUpdated={lastUpdated(latin, 'news')}
      service="news"
      script={latin}
      dir="ltr"
    />,
  );
});

describe('MostReadRank', () => {
  shouldMatchSnapshot(
    'should render ltr correctly',
    <MostReadRank service="news" script={latin}>
      5
    </MostReadRank>,
  );

  shouldMatchSnapshot(
    'should render ltr with double digits correctly',
    <MostReadRank service="news" script={latin}>
      10
    </MostReadRank>,
  );

  shouldMatchSnapshot(
    'should render rtl correctly',
    <MostReadRank service="persian" script={arabic}>
      ۲
    </MostReadRank>,
  );
});
