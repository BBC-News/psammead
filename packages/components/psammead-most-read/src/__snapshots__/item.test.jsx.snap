// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MostReadItemWrapper should render ltr correctly with 10 items 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 0;
  padding: 0;
}

.c3 {
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 300;
  font-style: normal;
  font-size: 2rem;
  line-height: 2.25rem;
  color: #B80000;
  margin: 0;
  padding: 0;
  float: left;
}

.c5 {
  font-size: 1.125rem;
  line-height: 1.375rem;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:hover,
.c5:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c7 {
  padding-bottom: 1.5rem;
  padding-left: 1rem;
  padding-right: 0;
}

.c4 {
  padding-bottom: 1.5rem;
  padding-left: 1rem;
  padding-right: 0;
}

.c8 {
  padding-bottom: 1.5rem;
  padding-left: 1rem;
  padding-right: 0;
}

.c11 {
  padding-bottom: 1.5rem;
  padding-left: 1rem;
  padding-right: 0;
}

.c13 {
  padding-bottom: 1.5rem;
  padding-left: 1rem;
  padding-right: 0;
}

.c14 {
  padding-bottom: 1.5rem;
  padding-left: 1rem;
  padding-right: 0;
}

@media (max-width:14.9375rem) {
  .c0 {
    width: calc(100%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c0 {
    width: calc(100%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c0 {
    width: calc(50%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c0 {
    width: calc(50%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    width: calc(50%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:80rem) {
  .c0 {
    width: calc(20%);
    display: inline-block;
    vertical-align: top;
  }
}

@supports (display:grid) {
  .c0 {
    display: block;
  }
}

@media (max-width:14.9375rem) {
  .c2 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c2 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c2 {
    min-width: auto;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c2 {
    min-width: auto;
  }
}

@media (min-width:80rem) {
  .c2 {
    min-width: auto;
  }
}

@media (max-width:14.9375rem) {
  .c6 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c6 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c6 {
    min-width: 2.5rem;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c6 {
    min-width: 4rem;
  }
}

@media (min-width:80rem) {
  .c6 {
    min-width: auto;
  }
}

@media (max-width:14.9375rem) {
  .c9 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c9 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c9 {
    min-width: auto;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c9 {
    min-width: auto;
  }
}

@media (min-width:80rem) {
  .c9 {
    min-width: 4.25rem;
  }
}

@media (max-width:14.9375rem) {
  .c10 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c10 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c10 {
    min-width: 2.5rem;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c10 {
    min-width: 4rem;
  }
}

@media (min-width:80rem) {
  .c10 {
    min-width: auto;
  }
}

@media (max-width:14.9375rem) {
  .c12 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c12 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c12 {
    min-width: auto;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c12 {
    min-width: auto;
  }
}

@media (min-width:80rem) {
  .c12 {
    min-width: auto;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c3 {
    font-size: 2.5rem;
    line-height: 2.75rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 3.5rem;
    line-height: 3.75rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c5 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c5 {
    font-size: 1.5rem;
    line-height: 1.75rem;
  }
}

@media (min-width:25rem) {
  .c7 {
    padding-right: 1.3rem;
  }
}

@media (min-width:37.5rem) {
  .c7 {
    padding-right: 1.8rem;
  }
}

@media (min-width:80rem) {
  .c7 {
    padding-right: 1.5rem;
  }
}

@media (min-width:25rem) {
  .c4 {
    padding-right: 0rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    padding-right: 0rem;
  }
}

@media (min-width:80rem) {
  .c4 {
    padding-right: 1.5rem;
  }
}

@media (min-width:25rem) {
  .c8 {
    padding-right: 1.3rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    padding-right: 1.8rem;
  }
}

@media (min-width:80rem) {
  .c8 {
    padding-right: 0rem;
  }
}

@media (min-width:25rem) {
  .c11 {
    padding-right: 1.3rem;
  }
}

@media (min-width:37.5rem) {
  .c11 {
    padding-right: 1.8rem;
  }
}

@media (min-width:80rem) {
  .c11 {
    padding-right: 1.5rem;
  }
}

@media (min-width:25rem) {
  .c13 {
    padding-right: 0rem;
  }
}

@media (min-width:37.5rem) {
  .c13 {
    padding-right: 0rem;
  }
}

@media (min-width:80rem) {
  .c13 {
    padding-right: 1.5rem;
  }
}

@media (min-width:25rem) {
  .c14 {
    padding-right: 0rem;
  }
}

@media (min-width:37.5rem) {
  .c14 {
    padding-right: 0rem;
  }
}

@media (min-width:80rem) {
  .c14 {
    padding-right: 0rem;
  }
}

<div>
  <div
    as="li"
    class="c0"
    dir="ltr"
  >
    <li
      class="c1"
    >
      <div
        class="c2"
        dir="ltr"
      >
        <span
          class="c3"
          dir="ltr"
        >
          1
        </span>
      </div>
      <div
        class="c4"
        dir="ltr"
      >
        <a
          class="c5"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </li>
  </div>
  <div
    as="li"
    class="c0"
    dir="ltr"
  >
    <li
      class="c1"
    >
      <div
        class="c6"
        dir="ltr"
      >
        <span
          class="c3"
          dir="ltr"
        >
          2
        </span>
      </div>
      <div
        class="c7"
        dir="ltr"
      >
        <a
          class="c5"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </li>
  </div>
  <div
    as="li"
    class="c0"
    dir="ltr"
  >
    <li
      class="c1"
    >
      <div
        class="c2"
        dir="ltr"
      >
        <span
          class="c3"
          dir="ltr"
        >
          3
        </span>
      </div>
      <div
        class="c4"
        dir="ltr"
      >
        <a
          class="c5"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </li>
  </div>
  <div
    as="li"
    class="c0"
    dir="ltr"
  >
    <li
      class="c1"
    >
      <div
        class="c6"
        dir="ltr"
      >
        <span
          class="c3"
          dir="ltr"
        >
          4
        </span>
      </div>
      <div
        class="c8"
        dir="ltr"
      >
        <a
          class="c5"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </li>
  </div>
  <div
    as="li"
    class="c0"
    dir="ltr"
  >
    <li
      class="c1"
    >
      <div
        class="c9"
        dir="ltr"
      >
        <span
          class="c3"
          dir="ltr"
        >
          5
        </span>
      </div>
      <div
        class="c4"
        dir="ltr"
      >
        <a
          class="c5"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </li>
  </div>
  <div
    as="li"
    class="c0"
    dir="ltr"
  >
    <li
      class="c1"
    >
      <div
        class="c10"
        dir="ltr"
      >
        <span
          class="c3"
          dir="ltr"
        >
          6
        </span>
      </div>
      <div
        class="c11"
        dir="ltr"
      >
        <a
          class="c5"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </li>
  </div>
  <div
    as="li"
    class="c0"
    dir="ltr"
  >
    <li
      class="c1"
    >
      <div
        class="c12"
        dir="ltr"
      >
        <span
          class="c3"
          dir="ltr"
        >
          7
        </span>
      </div>
      <div
        class="c13"
        dir="ltr"
      >
        <a
          class="c5"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </li>
  </div>
  <div
    as="li"
    class="c0"
    dir="ltr"
  >
    <li
      class="c1"
    >
      <div
        class="c10"
        dir="ltr"
      >
        <span
          class="c3"
          dir="ltr"
        >
          8
        </span>
      </div>
      <div
        class="c11"
        dir="ltr"
      >
        <a
          class="c5"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </li>
  </div>
  <div
    as="li"
    class="c0"
    dir="ltr"
  >
    <li
      class="c1"
    >
      <div
        class="c12"
        dir="ltr"
      >
        <span
          class="c3"
          dir="ltr"
        >
          9
        </span>
      </div>
      <div
        class="c14"
        dir="ltr"
      >
        <a
          class="c5"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </li>
  </div>
  <div
    as="li"
    class="c0"
    dir="ltr"
  >
    <li
      class="c1"
    >
      <div
        class="c10"
        dir="ltr"
      >
        <span
          class="c3"
          dir="ltr"
        >
          10
        </span>
      </div>
      <div
        class="c11"
        dir="ltr"
      >
        <a
          class="c5"
          href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
        >
          Could a computer ever create better art than a human?
        </a>
      </div>
    </li>
  </div>
</div>
`;

exports[`MostReadItemWrapper should render rtl correctly with 10 items 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 0;
  padding: 0;
}

.c3 {
  font-family: "BBC Nassim Persian",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 2rem;
  line-height: 2.25rem;
  color: #B80000;
  margin: 0;
  padding: 0;
  float: right;
}

.c5 {
  font-size: 1.5rem;
  line-height: 2rem;
  font-family: "BBC Nassim Persian",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 700;
  font-style: normal;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:hover,
.c5:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c7 {
  padding-bottom: 1.5rem;
  padding-right: 1rem;
  padding-left: 0;
}

.c4 {
  padding-bottom: 1.5rem;
  padding-right: 1rem;
  padding-left: 0;
}

.c8 {
  padding-bottom: 1.5rem;
  padding-right: 1rem;
  padding-left: 0;
}

.c11 {
  padding-bottom: 1.5rem;
  padding-right: 1rem;
  padding-left: 0;
}

.c13 {
  padding-bottom: 1.5rem;
  padding-right: 1rem;
  padding-left: 0;
}

.c14 {
  padding-bottom: 1.5rem;
  padding-right: 1rem;
  padding-left: 0;
}

@media (max-width:14.9375rem) {
  .c0 {
    width: calc(100%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c0 {
    width: calc(100%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c0 {
    width: calc(50%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c0 {
    width: calc(50%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    width: calc(50%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:80rem) {
  .c0 {
    width: calc(20%);
    display: inline-block;
    vertical-align: top;
  }
}

@supports (display:grid) {
  .c0 {
    display: block;
  }
}

@media (max-width:14.9375rem) {
  .c2 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c2 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c2 {
    min-width: auto;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c2 {
    min-width: auto;
  }
}

@media (min-width:80rem) {
  .c2 {
    min-width: auto;
  }
}

@media (max-width:14.9375rem) {
  .c6 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c6 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c6 {
    min-width: 2.5rem;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c6 {
    min-width: 4rem;
  }
}

@media (min-width:80rem) {
  .c6 {
    min-width: auto;
  }
}

@media (max-width:14.9375rem) {
  .c9 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c9 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c9 {
    min-width: auto;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c9 {
    min-width: auto;
  }
}

@media (min-width:80rem) {
  .c9 {
    min-width: 4.25rem;
  }
}

@media (max-width:14.9375rem) {
  .c10 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c10 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c10 {
    min-width: 2.5rem;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c10 {
    min-width: 4rem;
  }
}

@media (min-width:80rem) {
  .c10 {
    min-width: auto;
  }
}

@media (max-width:14.9375rem) {
  .c12 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c12 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c12 {
    min-width: auto;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c12 {
    min-width: auto;
  }
}

@media (min-width:80rem) {
  .c12 {
    min-width: auto;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c3 {
    font-size: 2.5rem;
    line-height: 2.75rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 3.5rem;
    line-height: 3.75rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c5 {
    font-size: 1.625rem;
    line-height: 2.25rem;
  }
}

@media (min-width:37.5rem) {
  .c5 {
    font-size: 1.75rem;
    line-height: 2.25rem;
  }
}

@media (min-width:25rem) {
  .c7 {
    padding-left: 1.3rem;
  }
}

@media (min-width:37.5rem) {
  .c7 {
    padding-left: 1.8rem;
  }
}

@media (min-width:80rem) {
  .c7 {
    padding-left: 1.5rem;
  }
}

@media (min-width:25rem) {
  .c4 {
    padding-left: 0rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    padding-left: 0rem;
  }
}

@media (min-width:80rem) {
  .c4 {
    padding-left: 1.5rem;
  }
}

@media (min-width:25rem) {
  .c8 {
    padding-left: 1.3rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    padding-left: 1.8rem;
  }
}

@media (min-width:80rem) {
  .c8 {
    padding-left: 0rem;
  }
}

@media (min-width:25rem) {
  .c11 {
    padding-left: 1.3rem;
  }
}

@media (min-width:37.5rem) {
  .c11 {
    padding-left: 1.8rem;
  }
}

@media (min-width:80rem) {
  .c11 {
    padding-left: 1.5rem;
  }
}

@media (min-width:25rem) {
  .c13 {
    padding-left: 0rem;
  }
}

@media (min-width:37.5rem) {
  .c13 {
    padding-left: 0rem;
  }
}

@media (min-width:80rem) {
  .c13 {
    padding-left: 1.5rem;
  }
}

@media (min-width:25rem) {
  .c14 {
    padding-left: 0rem;
  }
}

@media (min-width:37.5rem) {
  .c14 {
    padding-left: 0rem;
  }
}

@media (min-width:80rem) {
  .c14 {
    padding-left: 0rem;
  }
}

<div>
  <div
    as="li"
    class="c0"
    dir="rtl"
  >
    <li
      class="c1"
    >
      <div
        class="c2"
        dir="rtl"
      >
        <span
          class="c3"
          dir="rtl"
        >
          ۱
        </span>
      </div>
      <div
        class="c4"
        dir="rtl"
      >
        <a
          class="c5"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </li>
  </div>
  <div
    as="li"
    class="c0"
    dir="rtl"
  >
    <li
      class="c1"
    >
      <div
        class="c6"
        dir="rtl"
      >
        <span
          class="c3"
          dir="rtl"
        >
          ۲
        </span>
      </div>
      <div
        class="c7"
        dir="rtl"
      >
        <a
          class="c5"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </li>
  </div>
  <div
    as="li"
    class="c0"
    dir="rtl"
  >
    <li
      class="c1"
    >
      <div
        class="c2"
        dir="rtl"
      >
        <span
          class="c3"
          dir="rtl"
        >
          ۳
        </span>
      </div>
      <div
        class="c4"
        dir="rtl"
      >
        <a
          class="c5"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </li>
  </div>
  <div
    as="li"
    class="c0"
    dir="rtl"
  >
    <li
      class="c1"
    >
      <div
        class="c6"
        dir="rtl"
      >
        <span
          class="c3"
          dir="rtl"
        >
          ۴
        </span>
      </div>
      <div
        class="c8"
        dir="rtl"
      >
        <a
          class="c5"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </li>
  </div>
  <div
    as="li"
    class="c0"
    dir="rtl"
  >
    <li
      class="c1"
    >
      <div
        class="c9"
        dir="rtl"
      >
        <span
          class="c3"
          dir="rtl"
        >
          ۵
        </span>
      </div>
      <div
        class="c4"
        dir="rtl"
      >
        <a
          class="c5"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </li>
  </div>
  <div
    as="li"
    class="c0"
    dir="rtl"
  >
    <li
      class="c1"
    >
      <div
        class="c10"
        dir="rtl"
      >
        <span
          class="c3"
          dir="rtl"
        >
          ۶
        </span>
      </div>
      <div
        class="c11"
        dir="rtl"
      >
        <a
          class="c5"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </li>
  </div>
  <div
    as="li"
    class="c0"
    dir="rtl"
  >
    <li
      class="c1"
    >
      <div
        class="c12"
        dir="rtl"
      >
        <span
          class="c3"
          dir="rtl"
        >
          ۷
        </span>
      </div>
      <div
        class="c13"
        dir="rtl"
      >
        <a
          class="c5"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </li>
  </div>
  <div
    as="li"
    class="c0"
    dir="rtl"
  >
    <li
      class="c1"
    >
      <div
        class="c10"
        dir="rtl"
      >
        <span
          class="c3"
          dir="rtl"
        >
          ۸
        </span>
      </div>
      <div
        class="c11"
        dir="rtl"
      >
        <a
          class="c5"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </li>
  </div>
  <div
    as="li"
    class="c0"
    dir="rtl"
  >
    <li
      class="c1"
    >
      <div
        class="c12"
        dir="rtl"
      >
        <span
          class="c3"
          dir="rtl"
        >
          ۹
        </span>
      </div>
      <div
        class="c14"
        dir="rtl"
      >
        <a
          class="c5"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </li>
  </div>
  <div
    as="li"
    class="c0"
    dir="rtl"
  >
    <li
      class="c1"
    >
      <div
        class="c10"
        dir="rtl"
      >
        <span
          class="c3"
          dir="rtl"
        >
          ۱۰
        </span>
      </div>
      <div
        class="c11"
        dir="rtl"
      >
        <a
          class="c5"
          href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
        >
          در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
        </a>
      </div>
    </li>
  </div>
</div>
`;

exports[`MostReadLink should render ltr correctly 1`] = `
.c1 {
  font-size: 1.125rem;
  line-height: 1.375rem;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover,
.c1:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  padding-bottom: 1.5rem;
  padding-left: 1rem;
  padding-right: 0;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c1 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    font-size: 1.5rem;
    line-height: 1.75rem;
  }
}

@media (min-width:25rem) {
  .c0 {
    padding-right: 1.3rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    padding-right: 1.8rem;
  }
}

@media (min-width:80rem) {
  .c0 {
    padding-right: 1.5rem;
  }
}

<div
  class="c0"
  dir="ltr"
>
  <a
    class="c1"
    href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
  >
    Could a computer ever create better art than a human?
  </a>
</div>
`;

exports[`MostReadLink should render rtl correctly 1`] = `
.c1 {
  font-size: 1.5rem;
  line-height: 2rem;
  font-family: "BBC Nassim Persian",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 700;
  font-style: normal;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover,
.c1:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  padding-bottom: 1.5rem;
  padding-right: 1rem;
  padding-left: 0;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c1 {
    font-size: 1.625rem;
    line-height: 2.25rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    font-size: 1.75rem;
    line-height: 2.25rem;
  }
}

@media (min-width:25rem) {
  .c0 {
    padding-left: 1.3rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    padding-left: 1.8rem;
  }
}

@media (min-width:80rem) {
  .c0 {
    padding-left: 1.5rem;
  }
}

<div
  class="c0"
  dir="rtl"
>
  <a
    class="c1"
    href="https://www.bbc.com/persian/articles/cej3lzd5e0go"
  >
    در این جشنواره برای نخستین بار از کارگران افغانستان در شهر تهران تقدیر شد
  </a>
</div>
`;

exports[`MostReadLink should render with last updated date correctly 1`] = `
.c1 {
  font-size: 1.125rem;
  line-height: 1.375rem;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover,
.c1:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  padding-bottom: 1.5rem;
  padding-left: 1rem;
  padding-right: 0;
}

.c2 {
  font-size: 0.875rem;
  line-height: 1.125rem;
  color: #6E6E73;
  display: block;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c1 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    font-size: 1.5rem;
    line-height: 1.75rem;
  }
}

@media (min-width:25rem) {
  .c0 {
    padding-right: 1.3rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    padding-right: 1.8rem;
  }
}

@media (min-width:80rem) {
  .c0 {
    padding-right: 1.5rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 0.8125rem;
    line-height: 1rem;
  }
}

<div
  class="c0"
  dir="ltr"
>
  <a
    class="c1"
    href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
  >
    Could a computer ever create better art than a human?
  </a>
  <time
    class="c2"
    datetime="2019-03-01T14:00+00:00"
  >
    Last updated: 5th November 2016
  </time>
</div>
`;
